src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1+Builder.cs

```diff
@@ -307,11 +300,7 @@ public void AddRange(T[] items, int length)
                 var offset = this.Count;
                 this.Count += length;
 
-                var nodes = _elements;
-                for (int i = 0; i < length; i++)
-                {
-                    nodes[offset + i] = items[i];
-                }
+                Array.Copy(items, 0, _elements, offset, length);
```

Comment:
This is going to be a behavioral change in the case of some invalid input, as an ArgumentException will be generated instead of an IndexOutOfRangeExceptions in the case where length is larger than items.Length.